tests/Crypto/AesEncryptionStreamTestTrait.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Aws\Test\Crypto;
+
+use Aws\Crypto\Cipher\Cbc;
+use Aws\Crypto\Cipher\Ctr;
+use Aws\Crypto\Cipher\Ecb;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Psr7\CachingStream;
+
+trait AesEncryptionStreamTestTrait
+{
+    public function cartesianJoinInputCipherMethodProvider()
+    {
+        $toReturn = [];
+        $plainTexts = $this->unwrapProvider([$this, 'plainTextProvider']);
+        $ivs = $this->unwrapProvider([$this, 'cipherMethodProvider']);
+
+        for ($i = 0; $i < count($plainTexts); $i++) {
+            for ($j = 0; $j < count($ivs); $j++) {
+                $toReturn []= [$plainTexts[$i], clone $ivs[$j]];
+            }
+        }
+
+        return $toReturn;
+    }
+
+    public function cartesianJoinInputKeySizeProvider()
+    {
+        $toReturn = [];
+        $plainTexts = $this->unwrapProvider([$this, 'plainTextProvider']);
+        $keySizes = $this->unwrapProvider([$this, 'keySizeProvider']);
+
+        for ($i = 0; $i < count($plainTexts); $i++) {
+            for ($j = 0; $j < count($keySizes); $j++) {
+                $toReturn []= [
+                    $plainTexts[$i],
+                    $keySizes[$j],
+                ];
+            }
+        }
+
+        return $toReturn;
+    }
+
+    public function cipherMethodProvider()
+    {
+        $toReturn = [];
+        foreach ($this->unwrapProvider([$this, 'keySizeProvider']) as $keySize) {
+            $toReturn []= [new Cbc(
+                openssl_random_pseudo_bytes(openssl_cipher_iv_length('aes-256-cbc')),
+                $keySize
+            )];
+        }
+
+        return $toReturn;
+    }
+
+    public function seekableCipherMethodProvider()
+    {
+        return array_filter($this->cipherMethodProvider(), function (array $args) {
+            return !($args[0] instanceof Cbc);
+        });
+    }
+
+    public function keySizeProvider()
+    {
+        return [
+            [128],
+            [192],
+            [256],
+        ];
+    }
+
+    public function plainTextProvider() {
+        return [
+            [Psr7\stream_for('The rain in Spain falls mainly on the plain.')],
+            [Psr7\stream_for('Ø¯Ø³Øªâ€ŒÙ†ÙˆØ´ØªÙ‡â€ŒÙ‡Ø§ Ù†Ù…ÛŒâ€ŒØ³ÙˆØ²Ù†Ø¯')],
```

Comment:
I believe this is a copy-paste error. IIRC, this case (and the following one) were meant to exercise round-tripping of multibyte characters, e.g.:

```php
[Psr7\stream_for('دست‌نوشته‌ها نمی‌سوزند')],
[Psr7\stream_for('Рукописи не горят')],
```